py/server/deephaven/table.py

```diff
@@ -503,6 +504,110 @@ def meta_table(self) -> Table:
     def j_object(self) -> jpy.JType:
         return self.j_table
 
+    def iter_dict(self, cols: Optional[Union[str, Sequence[str]]] = None) -> Generator[Dict[str, Any], None, None]:
+        """ Returns a generator that reads one row at a time from the table into a dictionary. The dictionary is a map
+        of column names to scalar values of the column data type.
+
+        If the table is refreshing and no update graph locks are currently being held, the generator will try to acquire
+        the shared lock of the update graph before reading the table data. This provides a consistent view of the data.
+        The side effect of this is that the table will not be able to refresh while the table is being iterated on.
+        Additionally, the generator internally maintains a fill context. The auto acquired shared lock and the fill
+        context will be released after the generator is destroyed. That can happen implicitly when the generator
+        is used in a for-loop. When the generator is not used in a for-loop, to prevent resource leaks, it must be closed
+        after use by either (1) by setting it to None, (2) using the del statement, or (3) calling the close() method on it.
+
+        Args:
+            cols (Optional[Union[str, Sequence[str]]]): The columns to read. If None, all columns are read.
+
+        Returns:
+            A generator that yields a dictionary of column names to scalar values.
+
+        Raises:
+            ValueError
+        """
+        from deephaven._table_reader import _table_reader_dict # to prevent circular import
+        return _table_reader_dict(self, cols)
+
+    def iter_tuple(self, cols: Optional[Union[str, Sequence[str]]] = None, tuple_name: str = 'Deephaven') -> Generator[Tuple[Any, ...], None, None]:
```

Comment:
I think we still need to give user option to specify chunk_size.

Reply:
I disagree. Since this is row-based iteration,  providing a chunk_size parameter even if it is optional is confusing.

Reply:
All row-based iteration, ideally, would be chunk based; just because the user wants to consume it in a row-at-a-time fashion doesn't mean that we shouldn't give them the same amount of underlying control they could otherwise get by using the explicitly chunk-based interface. The first thing I would do if I was an end user and you didn't provide me an option to specify chunk size would be to wrap the chunk based impls which do provide `chunk_size` and re-expose it in a row-at-a-time fashion.